bluez-async/src/adapter.rs

```diff
@@ -27,3 +30,101 @@ impl Display for AdapterId {
         )
     }
 }
+
+/// Information about a Bluetooth adapter on the system.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct AdapterInfo {
+    /// An opaque identifier for the adapter. This can be used to perform operations on it.
+    pub id: AdapterId,
+    /// The MAC address of the adapter.
+    pub mac_address: MacAddress,
+    /// The type of MAC address the adapter uses.
+    pub address_type: AddressType,
+    /// The Bluetooth system hostname.
+    pub name: String,
+    /// The Bluetooth friendly name. This defaults to the system hostname.
+    pub alias: String,
+    /// Whether the adapter is currently turned on.
+    pub powered: bool,
+    /// Whether the adapter is currently discovering devices.
+    pub discovering: bool,
+}
+
+impl AdapterInfo {
+    pub(crate) fn from_properties(
+        id: AdapterId,
+        adapter_properties: OrgBluezAdapter1Properties,
+    ) -> Result<AdapterInfo, BluetoothError> {
+        let mac_address = adapter_properties
+            .address()
+            .ok_or_else(|| BluetoothError::RequiredPropertyMissing("Address".to_string()))?;
```

Comment:
could this prop take a &'static str or have a constructor that takes &str?

Reply:
The error type you mean? Done.